### PR TITLE
[DO-NOT-MERGE] LAPI for supporting 3rd party IAM integration [DPP-1142]

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/sandbox-on-x/reference.conf [hunk 1]
```diff
@@ -272,7 +272,8 @@ ledger {
         }
       }
 
-      authentication {
+      # We make it a list, but will allow for at most two entries, each with a different role.
+      authentication = [{
```
Comment (pbatko-da) [inline ledger/sandbox-on-x/reference.conf:276]: authetication-domains {
  __default = { **authecaation } 
 }

File: ledger/sandbox-on-x/reference.conf [hunk 2]
```diff
@@ -342,6 +343,56 @@ ledger {
         #}
       }
 
+      # New style config sample:
+      authentication-domains  {
```
Comment (skisel-da) [inline ledger/sandbox-on-x/reference.conf:347]: I would probably put `domains` inside `authentication` section, i.e. the path would `be authentication.domain`

File: ledger/sandbox-on-x/reference.conf [hunk 3]
```diff
@@ -342,6 +343,56 @@ ledger {
...
+        # 'hub' is an arbitrary name given to an authetication domain
+        hub {
+            # 'authentication' section - the same as the existing 'authentication' above
+            authentication {
```
Comment (skisel-da) [inline ledger/sandbox-on-x/reference.conf:351]: we would need a better name here if we put "domains" inside "authentication"

File: ledger/sandbox-on-x/reference.conf [hunk 4]
```diff
@@ -342,6 +343,56 @@ ledger {
         #}
       }
 
+      # New style config sample:
+      authentication-domains  {
+        # 'hub' is an arbitrary name given to an authetication domain
+        hub {
```
Comment (skisel-da) [inline ledger/sandbox-on-x/reference.conf:349]: I suspect we also need a way to disable the domain with `enabled=false`

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto [hunk 5]
```diff
@@ -138,6 +138,13 @@ message User {
     // Optional
     // Modifiable
     ObjectMeta metadata = 4;
+
+    // The name of an authentication domain this user belongs to.
+    // An authentication domain specifies how the participant authenticates the authentication tokens issued for this user.
+    //
+    // Optional. The empty string value is equivalent to specifying the 'default' authentication domain.
+    // Modifiable
+    string auth_domain_name = 5;
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto:147]:
```suggestion
    string authn_domain_name = 5;
```
following https://www.cloudflare.com/learning/access-management/authn-vs-authz/#

**Review by meiersi-da — COMMENTED** [review]
Thank you @pbatko-da -- looks good to me from a L.API semantics perspective.

**Pull request closed without merging**